File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;
        for(int i = 0;i< nums.size();i++){
            int remainder = target - nums[i];
            if  (map.count(remainder)){
                return {map[remainder],i};
            }
            map[nums[i]] = i;
        }
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        sum = 0
        path = []
        def backtrack (i, path, sum):
            if sum == target: 
                res.append(path.copy())
                return
            
            
            if sum >= target or i >= len(nums):
                return
            path.append(nums[i])
            
            backtrack(i,path,sum + nums[i])
            path.pop()
            backtrack(i+1,path,sum)

        backtrack(0,[],0)
        return res